backend/routes/user.py

```diff
@@ -46,4 +51,33 @@ async def create_account(user: CreateUserSchema= Body(...)):
 
     return {'status_code':200, 'message': 'User saved successfully'}
 
+# Use OAuth2PasswordBearer for authorization
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/verify_otp")
+
+@router.post("/api/v1/verify_otp")
+async def verify_otp(vphone_number: str, votp: int, is_creation: bool):
+    # check expiration of entered OTP
+    curr_time = datetime.datetime.now()
+    otp_gen_time = database.otp_mapping.find_one({'phone_number': vphone_number, 'otp': votp},{'timestamp': 1,'_id': 0})
+    valid_time = otp_gen_time['timestamp'] + datetime.timedelta(seconds=60)
+    is_otp_valid = curr_time <= valid_time
+
```

Comment:
what would happen in case the  OTP entered is incorrect.  line 61 wont find any record and subsequent lines wont work as intended. Please check the expiration of OTP only when OTP entered is correct.

Reply:
I got the point and  will make changes accordingly